include/oneapi/mkl/dft/detail/commit_impl.hpp

```diff
@@ -28,35 +28,37 @@
 #endif
 
 #include "oneapi/mkl/detail/backends.hpp"
+#include "types_impl.hpp"
 
 namespace oneapi {
 namespace mkl {
 namespace dft {
 namespace detail {
 
+template <precision prec, domain dom>
```

Comment:
Why does `commit_impl` need these templates now?

Reply:
Because I introduced a pure virtual function `commit` that takes a `dft_values<prec,dom>&` argument, meaning the class needs to know the type of `prec` and `dom`.